app/models/user.rb

```diff
@@ -0,0 +1,30 @@
+class User < ApplicationRecord
+  has_many :reviews, dependent: :destroy
+  has_one :billing_address, as: :addressable, dependent: :destroy
+  has_one :shipping_address, as: :addressable, dependent: :destroy
+  has_many :orders, dependent: :destroy
+  has_many :order_items, through: :orders
+  has_many :books, through: :order_items
+
+  accepts_nested_attributes_for :billing_address
```

Comment:
would be better to have separate logic for address updating, `accepts_nested_attributes_for` is a bad practice